pystac/asset.py

```diff
@@ -0,0 +1,248 @@
+from copy import copy
+
+from pystac import STACError
+from pystac.utils import (is_absolute_href, make_absolute_href, make_relative_href)
+
+
+class AssetsMixin:
+    """Mixin class for classes that contain assets.
+
+    Implementing classes must have a `assets` attribute that is
+    a dictionary of Asset objects.
+    """
+    def get_assets(self):
+        """Get this object's assets.
+
+        Returns:
+            Dict[str, Asset]: A copy of the dictionary of this object's assets.
+        """
+        return dict(self.assets.items())
+
+    def add_asset(self, key, asset):
+        """Adds an Asset to this object.
+
+        Args:
+            key (str): The unique key of this asset.
+            asset (Asset): The Asset to add.
+        """
+        asset.set_owner(self)
+        self.assets[key] = asset
+        return self
+
+    def make_asset_hrefs_relative(self):
+        """Modify each asset's HREF to be relative to this object's self HREF.
+
+        Returns:
+            STACObjectWithAssets: self
```

Comment:
docstring references to removed type

Reply:
I removed all references to the STACObjectWithAssets class